xlators/protocol/client/src/client-rpc-fops.c

```diff
@@ -5100,6 +5112,18 @@ client3_3_lk(call_frame_t *frame, xlator_t *this, void *data)
             client_is_setlk(local->cmd)) {
             client_add_lock_for_recovery(local->fd, args->flock, &local->owner,
                                          local->cmd);
+        } else if ((op_errno == EBADFD) && conf->strict_locks &&
+                   (args->flock->l_type == F_WRLCK) &&
+                   (local->cmd == F_GETLK)) {
+            ret = dict_get_int32(args->xdata, "fix-open", &fix_open);
```

Comment:
I think this dict check alone is sufficient since only the afr code sets the key in the dict.

Reply:
Kept the local->cmd check since we need to check whether fix-open has called this or not and to avoid doing dict_get in all other cases.

Reply:
Here we check ""fix-open" in the dict to set_strict_locks_in_dict() but in the call back path,  we are setting it unconditionally. Likewise in client4_0_lk()